src/processor/operator/copy/read_npy.cpp

```diff
@@ -0,0 +1,90 @@
+#include "processor/operator/copy/read_npy.h"
+
+#include "common/constants.h"
+#include "storage/in_mem_storage_structure/in_mem_column_chunk.h"
+
+namespace kuzu {
+namespace processor {
+
+void ReadNPYSharedState::countNumLines() {
+    uint8_t idx = 0;
+    uint64_t firstFileRows;
+    for (auto& filePath : filePaths) {
+        std::unique_ptr<storage::NpyReader> reader = make_unique<storage::NpyReader>(filePath);
+        numRows = reader->getNumRows();
+        if (idx == 0) {
+            firstFileRows = numRows;
+        }
+        auto tableType = tableSchema->getProperty(idx).dataType;
+        reader->validate(tableType, firstFileRows, tableSchema->tableName);
+        auto numBlocks = (uint64_t)((numRows + common::CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY) /
+                                    common::CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY);
+        std::vector<uint64_t> numRowsPerBlock(numRows,
```

Comment:
This line should be incorrect.
It should be `std::vector<uint64_t> numRowsPerBlock(numBlocks, common::CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY)`. Or 
```
        auto blockIdx = 0;
        std::vector<uint64_t> numRowsPerBlock(numBlocks);
        while (blockIdx < numBlocks) {
            numRowsPerBlock[blockIdx] = std::min(CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY,
                numRows - (blockIdx * CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY));
            blockIdx++;
        }
```